src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -191,7 +192,27 @@ bool concat_in_place_optimization::match(const program_node& concat_node,
             if (pred_l.format == format::b_fs_yx_fsv4 && (pred_l.feature() != 4 || concat_axis != 1))
                 return false;
         }
+
         if (pred.first->get_preferred_impl_type() == impl_types::onednn) {
+            // Onednn requires memory pointers to be aligned at least at 64-bytes to avoid potential correctness issues.
+            // https://uxlfoundation.github.io/oneDNN/dev_guide_c_and_cpp_apis.html#intel-r-processor-graphics-and-xe-architecture-graphics
+            if (!concat_node.is_dynamic() || is_runtime) {
+                if (onednn_byte_offset % 64 != 0)
+                    return false;
+
+                size_t element_size = 1;
+                if (concat_axis == 1) {
+                    element_size *= pred_l.feature();
+                } else {
+                    return false;
+                }
+                for (size_t i = 0; i < pred_l.get_spatial_rank(); ++i) {
+                    element_size *= pred_l.spatial(i);
+                }
+                auto byte_size = element_size * ov::element::Type(pred_l.data_type).bitwidth() / 8;
```

Comment:
isn't it enough to use `pred_l.get_bytes_count()`? (but this will include batch, too)

Reply:
but the assumption here is that onednn will support b=1 case. We have comment `// If sibling is using onednn impl and batch > 1, the onednn impl cannot process the implicit concat'ed buffer.` So it is OK to include batch axis.

Reply:
Okay, applied it.